KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,110 +1,327 @@
-# chapter 14 output file into a pdf
-
-
-def prepareForPDF(chap):
-    footmiscIndex = 0
+# Path to data directory
+DATA_DIR = os.path.dirname(os.path.realpath(__file__)) + "/../data/"
+
+# holds all subsections in each chapter section along with the what holds
+
+# w, h = 2, 100
+# sorter_check = [[0 for _ in range(w)] for __ in range(h)]
+
+
+def chap1placer(chap1, kls, klsaddparas):
+    chapterstart = True
+    index = 0
+    kls_sub_chap1 = []
+    kls_header_chap1 = []
+    for item in kls:
+        index += 1
+        line = str(item)
+        if "\\subsection" in line:
+            temp = line[line.find(" ", 12) + 1: line.find("}", 12)+1]  # get just the name (like mathpeople)
+        if chapterstart and ("\\paragraph{" in line or "\\subsubsection*{" in line):
+            for item in klsaddparas:
+                if index < item:
+                    klsloc = klsaddparas.index(item)
+                    break
+            t = ''.join(kls[index-1: klsaddparas[klsloc]])
+            kls_sub_chap1.append(t)  # append the whole paragraph, pray every paragraph ends with a % comment
+            kls_header_chap1.append(temp)  # append the name of subsection
+            break
+    intro_to_add = ''.join(kls_sub_chap1)
+    chap1 = chap1[:len(chap1)-1]
+    chap1 = ''.join(chap1)
+    total_chap1 = chap1 + "\\paragraph{\\bf KLS Addendum: Generalities}" + intro_to_add + "\\end{document}"
+    return total_chap1
+
+
+def extraneous_section_deleter(list):
+    """
+    Removes sections that are irrelevant and will slow or confuse the program
+
+    :param list:
+    :return: Extraneous name free output
+    """
+    return [item for item in list if "reference" not in item.lower() and "limit relation" not in item.lower()
+            and "symmetry" not in item.lower() and " hypergeometric representation" not in item.lower()
+            and "hypergeometric representation " not in item.lower()]
+
+
+def new_keywords(kls, kls_list):
+    """
+    This section checks through the Addendum and identifies words that are valid keywords of sections
+    It then takes that list and removes duplicates, as well as printing the output to another file.
+
+    :param kls: The addendum that provides the words.
+    :return: Inputs results into a global list.
+    """
+
+    kls_list_chap = []
+    for item in kls:
+        if "paragraph{" in item or "subsubsection*{" in item:
+            kls_list.append(item[item.find("{") + 1: len(item) - 2])
+    kls_list = extraneous_section_deleter(kls_list)
+
+    kls_list.append("Limit Relation")
+    for item in kls_list:
+        if item not in kls_list_chap:
+            kls_list_chap.append(item)
+    kls_list_chap[kls_list_chap.index("Special value")] = "Symmetry"
+    kls_list_chap.append("Special value")
+    w = 2
+    h = len(kls_list_chap)+1
+    sorter_check = [[0 for _ in range(w)] for __ in range(h)]
+    return kls_list_chap, sorter_check
```

Comment:
I don't know what's not as intended - your docstring or what the function actually does, because the docstring suggests that this doesn't return anything and rather modifies a global